mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -76,7 +79,10 @@ private void loadConfigAndCheckIfInstallationShouldBeSkipped(MavenSession sessio
             return;
         }
 
-        File projectDirectory = session.getTopLevelProject().getModel().getProjectDirectory();
+        final File multiModuleProjectDirectory = session.getRequest().getMultiModuleProjectDirectory();
+        final File executionDir = session.getTopLevelProject().getModel().getProjectDirectory();
+        final File projectDirectory = findFirstDirectoryWithConfigFile(executionDir, multiModuleProjectDirectory);
```

Comment:
When the configuration file is loaded this message is printed on the stdout:
`Smart Testing Extension - Using configuration from smart-testing.yaml`
Now, when the config file can be loaded from a different directory it would be good to have the full path to the file visible in the log message.

Reply:
Full path or relative to the project folder (whichever is easier to implement for the time being), but I agree we should have it.

Reply:
But this made me thinking - shouldn't we have hierarchical lookup here? e.g. I might have a config file defining e.g. strategies, but the mode we have globally enabled on the root level. Do we handle such a case now @dipak-pawar?

Reply:
No we don't handle such case.

Reply:
> shouldn't we have hierarchical lookup here?

then we'll miss scenario where we have config file defined in root directory of project & user is trying to run test from sub-module directory. Then as in sub-module directory we don't have config file it'll take default values which is not correct.

Reply:
I thought that was the idea overall, so user can have only the relevant bits configured in the sub-module. But I understand current solution is all-or-nothing? 

> then we'll miss scenario where we have config file defined in root directory of project & user is trying to run test from sub-module directory. Then as in sub-module directory we don't have config file it'll take default values which is not correct.

I am not sure I understand your point clearly. The idea might be the same as with hierarchy of poms. But if that's not how it is implemented we have to make sure we document it properly and if we agree this is the right thing to do - create a task.

Reply:
Will be discussed offline with @dipak-pawar.